app/uk/gov/hmrc/debttransformationstub/repositories/TTPRequestsRepository.scala

```diff
@@ -17,66 +17,56 @@
 package uk.gov.hmrc.debttransformationstub.repositories
 
 import com.google.inject.{ ImplementedBy, Singleton }
-import play.api.libs.json.{ JsBoolean, JsString }
-import play.modules.reactivemongo.ReactiveMongoComponent
-import reactivemongo.api.commands.WriteResult
-import reactivemongo.api.indexes.{ Index, IndexType }
-import reactivemongo.bson.BSONObjectID
+import org.mongodb.scala.model.Indexes.ascending
+import org.mongodb.scala.model.{ Filters, IndexModel, IndexOptions }
+import org.mongodb.scala.result.{ DeleteResult, InsertOneResult }
 import uk.gov.hmrc.debttransformationstub.models.RequestDetail
-import uk.gov.hmrc.mongo.ReactiveRepository
-import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 
 import javax.inject.Inject
 import scala.concurrent.{ ExecutionContext, Future }
 
 @ImplementedBy(classOf[TTPRequestsRepositoryImpl])
 trait TTPRequestsRepository {
-  def deleteTTPRequest(requestId: String): Future[WriteResult]
+  def deleteTTPRequest(requestId: String): Future[DeleteResult]
   def findRequestDetails(): Future[List[RequestDetail]]
   def findUnprocessedRequestDetails(): Future[List[RequestDetail]]
   def getByRequestId(id: String): Future[Option[RequestDetail]]
   def getResponseByRequestId(id: String): Future[Option[RequestDetail]]
-  def insertRequestsDetails(requestDetail: RequestDetail): Future[WriteResult]
+  def insertRequestsDetails(requestDetail: RequestDetail): Future[InsertOneResult]
 
 }
 
 @Singleton
-class TTPRequestsRepositoryImpl @Inject() (implicit mongo: ReactiveMongoComponent, ec: ExecutionContext)
-    extends ReactiveRepository[RequestDetail, BSONObjectID](
+class TTPRequestsRepositoryImpl @Inject()(mongo: MongoComponent)(implicit ec: ExecutionContext)
+    extends PlayMongoRepository[RequestDetail](
+      mongo,
       "ttp-requests",
-      mongo.mongoConnector.db,
       RequestDetail.requestDetailFormat,
-      ReactiveMongoFormats.objectIdFormats
+      indexes =
+        Seq(IndexModel(ascending("referenceId"), IndexOptions().name("referenceIdUnique").unique(true).sparse(true)))
```

Comment:
Not sure the .name(referenceIdUnique) is needed? Why isn't the name left as referenceId?